third_party/move/move-compiler-v2/src/pipeline/livevar_analysis_processor.rs

```diff
@@ -321,14 +327,26 @@ impl<'a> CopyTransformation<'a> {
     fn check_implicit_copy(
         &self,
         alive: &LiveVarInfoAtCodeOffset,
-        _id: AttrId,
+        id: AttrId,
         _is_updated: bool,
         temp: TempIndex,
     ) -> bool {
-        // TODO(#10723): insert ability check here
         // Notice we do allow copy of &mut. Those copies are checked for validity in
         // reference safety.
-        alive.after.contains_key(&temp)
+        let needed = alive.after.contains_key(&temp);
+        if needed {
+            let target = self.target();
+            check_copy(
```

Comment:
Here, line 386, and line 416 you have an identical expression which only needs `target` as a parameter.  Pull it out to a function, or reduce the number of parameters to `check_copy` to just target.

Reply:
done

Reply:
@brmataptos Yes, a variable can be dead and borrowed; this happens all the time. The problem is when the variable is dead *before and after* the program point, it's still borrowed. The former can only happen when the variable is written to at the program point, say `x = ...`, and `x` is never used later. But the latter says `x` is borrowed after the program point nevertheless.